node/src/components/storage/tests.rs

```diff
@@ -1144,3 +1153,112 @@ fn should_hard_reset() {
     check(0);
 }
+
+#[test]
+fn should_create_subdir_named_after_network() {
+    let harness = ComponentHarness::default();
+    let cfg = new_config(&harness);
+
+    let storage = Storage::new(
+        &WithDir::new(harness.tmp.path(), cfg.clone()),
+        None,
+        ProtocolVersion::from_parts(1, 0, 0),
+        false,
+        "test",
+    )
+    .unwrap();
+
+    let expected_path = cfg.path.join("test");
+
+    assert!(expected_path.exists());
+    assert_eq!(expected_path, storage.root_path());
+}
+
+#[test]
+fn should_not_move_nonexistent_files() {
+    let harness = ComponentHarness::default();
+    let cfg = new_config(&harness);
+    let file_names = ["temp.txt"];
+
+    let actual = should_move_storage_files_to_network_subdir(&cfg.path, &file_names);
+
+    assert_eq!(false, actual);
+}
+
+#[test]
+fn should_move_files_if_they_exist() {
+    let harness = ComponentHarness::default();
+    let cfg = new_config(&harness);
+    let file_names = ["temp1.txt", "temp2.txt", "temp3.txt"];
+
+    // Storage will create this in the constructor,
+    // doing this manually since we're not calling the constructor in this test.
+    fs::create_dir(cfg.path.clone()).unwrap();
+
+    // create empty files for testing.
+    File::create(cfg.path.join(file_names[0])).unwrap();
+    File::create(cfg.path.join(file_names[1])).unwrap();
+    File::create(cfg.path.join(file_names[2])).unwrap();
+
+    let actual = should_move_storage_files_to_network_subdir(&cfg.path, &file_names);
+
+    assert_eq!(true, actual);
+}
+
+#[test]
+fn should_not_move_files_if_some_missing() {
+    let harness = ComponentHarness::default();
+    let cfg = new_config(&harness);
+    let file_names = ["temp1.txt", "temp2.txt", "temp3.txt"];
+
+    // Storage will create this in the constructor,
+    // doing this manually since we're not calling the constructor in this test.
+    fs::create_dir(cfg.path.clone()).unwrap();
+
+    // create empty files for testing, but not all of the files.
+    File::create(cfg.path.join(file_names[1])).unwrap();
+    File::create(cfg.path.join(file_names[2])).unwrap();
+
+    let actual = should_move_storage_files_to_network_subdir(&cfg.path, &file_names);
+
+    assert_eq!(false, actual);
+}
+
+#[test]
+fn should_actually_move_specified_files() {
+    let harness = ComponentHarness::default();
+    let cfg = new_config(&harness);
+    let file_names = ["temp1.txt", "temp2.txt", "temp3.txt"];
+    let root = cfg.path;
+    let subdir = root.join("test");
+    let src_path1 = root.join(file_names[0]);
+    let src_path2 = root.join(file_names[1]);
+    let src_path3 = root.join(file_names[2]);
+    let dest_path1 = subdir.join(file_names[0]);
+    let dest_path2 = subdir.join(file_names[1]);
+    let dest_path3 = subdir.join(file_names[2]);
+
+    // Storage will create this in the constructor,
+    // doing this manually since we're not calling the constructor in this test.
+    fs::create_dir(root.clone()).unwrap();
```

Comment:
You could use `fs::create_dir_all()` here.  It probably doesn't matter that it doesn't return a `Result` since you're asserting the files exist before trying to move them.